sw/device/tests/otbn/otbn_ecdsa_p256_test.c

```diff
@@ -0,0 +1,262 @@
+                            uint8_t *signature_s) {
+  CHECK(otbn_ctx != NULL);
+
+  // Write input arguments.
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/32, msg, kOtbnVarMsg) ==
+        kOtbnOk);
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/32, secret_random_int_k,
+                               kOtbnVarK) == kOtbnOk);
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/32, private_key_d,
+                               kOtbnVarD) == kOtbnOk);
+
+  // Call OTBN to perform operation, and wait for it to complete.
+  CHECK(otbn_call_function(otbn_ctx, kOtbnAppP256EcdsaFuncSign) == kOtbnOk);
+  CHECK(otbn_busy_wait_for_done(otbn_ctx) == kOtbnOk);
+
+  // Read back results.
+  CHECK(otbn_copy_data_from_otbn(otbn_ctx, /*len_bytes=*/32, kOtbnVarR,
+                                 signature_r) == kOtbnOk);
+  CHECK(otbn_copy_data_from_otbn(otbn_ctx, /*len_bytes=*/32, kOtbnVarS,
+                                 signature_s) == kOtbnOk);
+}
+
+/**
+ * Verifies a message with ECDSA using the P-256 curve.
+ *
+ * @param otbn_ctx             The OTBN context object.
+ * @param msg                  The message to verify (32B).
+ * @param signature_s          The signature component s (the proof) (32B).
+ * @param public_key_x         The public key x-coordinate (32B).
+ * @param public_key_y         The public key y-coordinate (32B).
+ * @param[out] signature_r_out Recovered point r' (== R'.x). Provide a
+ *                             pre-allocated 32B buffer.
+ */
+static void p256_ecdsa_verify(otbn_t *otbn_ctx, const uint8_t *msg,
+                              const uint8_t *signature_s,
+                              const uint8_t *public_key_x,
+                              const uint8_t *public_key_y,
+                              uint8_t *signature_r_out) {
+  CHECK(otbn_ctx != NULL);
+
+  // Write input arguments.
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/32, msg, kOtbnVarMsg) ==
+        kOtbnOk);
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/32, signature_s,
+                               kOtbnVarS) == kOtbnOk);
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/32, public_key_x,
+                               kOtbnVarX) == kOtbnOk);
+  CHECK(otbn_copy_data_to_otbn(otbn_ctx, /*len_bytes=*/32, public_key_y,
+                               kOtbnVarY) == kOtbnOk);
+
+  // Call OTBN to perform operation, and wait for it to complete.
+  CHECK(otbn_call_function(otbn_ctx, kOtbnAppP256EcdsaFuncVerify) == kOtbnOk);
+  CHECK(otbn_busy_wait_for_done(otbn_ctx) == kOtbnOk);
+
+  // Read back results.
+  CHECK(otbn_copy_data_from_otbn(otbn_ctx, /*len_bytes=*/32, kOtbnVarRnd,
+                                 signature_r_out) == kOtbnOk);
+}
+
+/**
+ * Performs a ECDSA roundtrip test using the NIST P-256 curve.
+ *
+ * A roundtrip consists of three steps: Initialize OTBN, sign, and verify.
+ */
+static void test_ecdsa_p256_roundtrip(void) {
+  // message
```

Comment:
nit: the capitalization of these comments should probably be consistent (see `// Initialize` and friends at line 225 below).